gemrb/core/PathFinder.cpp

```diff
@@ -0,0 +1,415 @@
+	if (Orientation <= 4) {
+		xoff = -Orientation / 4.0;
+	} else if (Orientation <= 12) {
+		xoff = -1.0 + (Orientation - 4) / 4.0;
+	} else {
+		xoff = 1.0 - (Orientation - 12) / 4.0;
+	}
+
+	if (Orientation <= 8) {
+		yoff = 1.0 - Orientation / 4.0;
+	} else {
+		yoff = -1.0 + (Orientation - 8) / 4.0;
+	}
+
+	mult = 1.0 / std::max(std::fabs(xoff), std::fabs(yoff));
+
+	dest.x += Steps * mult * xoff + 0.5;
+	dest.y += Steps * mult * yoff + 0.5;
+
+	return GetLine(start, dest, 2, Orientation, flags);
+}
+
+PathNode *Map::GetLine(const Point &start, const Point &dest, int Speed, int Orientation, int flags) const
+{
+	PathNode *StartNode = new PathNode;
+	PathNode *Return = StartNode;
+	StartNode->Next = NULL;
+	StartNode->Parent = NULL;
+	StartNode->x = start.x;
+	StartNode->y = start.y;
+	StartNode->orient = Orientation;
+
+	int Count = 0;
+	int Max = Distance(start, dest);
+	for (int Steps = 0; Steps < Max; Steps++) {
+		Point p;
+		p.x = (ieWord) start.x + ((dest.x - start.x) * Steps / Max);
+		p.y = (ieWord) start.y + ((dest.y - start.y) * Steps / Max);
+
+		//the path ends here as it would go off the screen, causing problems
+		//maybe there is a better way, but i needed a quick hack to fix
+		//the crash in projectiles
+		if ((signed) p.x < 0 || (signed) p.y < 0) {
+			return Return;
+		}
+		if ((ieWord) p.x > Width * 16 || (ieWord) p.y > Height * 12) {
+			return Return;
+		}
+
+		if (!Count) {
+			StartNode->Next = new PathNode;
+			StartNode->Next->Parent = StartNode;
+			StartNode = StartNode->Next;
+			StartNode->Next = NULL;
+			Count = Speed;
+		} else {
+			Count--;
+		}
+
+		StartNode->x = p.x;
+		StartNode->y = p.y;
+		StartNode->orient = Orientation;
+		bool wall = GetBlocked(p.x / 16, p.y / 12) & (PATH_MAP_DOOR_IMPASSABLE | PATH_MAP_SIDEWALL);
+		if (wall) switch (flags) {
+			case GL_REBOUND:
+				Orientation = (Orientation + 8) & 15;
+				// TODO: recalculate dest (mirror it)
+				break;
+			case GL_PASS:
+				break;
+			default: //premature end
+				return Return;
+		}
+	}
+
+	return Return;
+}
+
+PathNode *Map::GetLine(const Point &p, int steps, unsigned int orient) const
+{
+	PathNode *step = new PathNode;
+	step->x = p.x + steps * SEARCHMAP_SQUARE_DIAGONAL * dxRand[orient];
+	step->y = p.y + steps * SEARCHMAP_SQUARE_DIAGONAL * dyRand[orient];
+	step->x = Clamp(step->x, 1u, (Width - 1) * 16);
+	step->y = Clamp(step->y, 1u, (Height - 1) * 12);
+	step->orient = GetOrient(Point(step->x, step->y), p);
+	step->Next = NULL;
+	step->Parent = NULL;
+	return step;
+}
+
+// Find a path from start to goal, ending at the specified distance from the
+// target (the goal must be in sight of the end, if PF_SIGHT is specified)
+PathNode *Map::FindPath(const Point &s, const Point &d, unsigned int size, unsigned int minDistance, int flags, const Actor *caller) const
+{
+	Log(DEBUG, "FindPath", "s = (%d, %d), d = (%d, %d), caller = %s, dist = %d, size = %d", s.x, s.y, d.x, d.y, caller ? caller->GetName(0) : "NULL", minDistance, size);
```

Comment:
Time to go or should we wait a bit?
```suggestion

```

Reply:
This is less frequent. I think we can leave it in for now, and if somebody complains about a spam of these it's either an issue or we can just remove it then. If nobody complains about it for, say, a month, then I guess it can go.